x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -25,10 +26,30 @@ import {
   InferencePipelineInferenceConfig,
 } from '../types/pipelines';
 
-export const TEXT_EXPANSION_TYPE = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
-export const TEXT_EXPANSION_FRIENDLY_TYPE = 'ELSER';
-export const ML_INFERENCE_PREFIX = 'ml.inference.';
-export const ELSER_MODEL_ID = '.elser_model_1';
+const ML_INFERENCE_PREFIX = 'ml.inference.';
+
+class InferenceModel {
+  public taskType: string;
+  public friendlyTaskType: string;
+  public modelId: string;
+
+  constructor() {
+    this.taskType = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
+    this.friendlyTaskType = 'ELSER';
+    this.modelId = '';
+  }
+
+  public setModelId(modelId: string): void {
+    this.modelId = modelId;
+  }
+}
+
+export const elserInferenceModel = new InferenceModel();
+
+export const configureElserInferenceModel = async (elasticModels: ElasticModels) => {
```

Comment:
```suggestion
export const configureElserInferenceModel = async (elasticModels: MlPluginStart['elasticModels']) => {
```

Reply:
Thank you for the tip. It does not seems to be used in the new version of the code.